src/main/java/de/muenchen/dave/services/IndexServiceUtils.java

```diff
@@ -86,13 +87,13 @@ public static List<Integer> getZaehljahre(List<Zaehlung> zs) {
      * @param zs Zaehlungen
      * @return aktuellste Zaehlung
      */
-    public static Zaehlung getLetzteZaehlung(List<Zaehlung> zs) {
+    public static Zaehlung getLetzteAktiveZaehlung(List<Zaehlung> zs) {
         zs.sort(Comparator.comparing(Zaehlung::getDatum));
-
-        if (!zs.isEmpty()) {
-            return zs.get(zs.size() - 1);
+        final List<Zaehlung> onlyActiveZaehlungen = zs.stream().filter(zaehlung -> Status.ACTIVE.name().equalsIgnoreCase(zaehlung.getStatus())).toList();
```

Comment:
Das if-else-Construct kann durch die Verwendung von Streams eingespart werden.

`zs.stream().filter(zaehlung -> Status.ACTIVE.name().equalsIgnoreCase(zaehlung.getStatus())).sort(Reverse ByDatum oder ähnliches).findFirst().orElse(do some logging and return null);`